ols/app/endpoints/feedback.py

```diff
@@ -1,33 +1,197 @@
+
+    Args:
+        feedback_id: The feedback ID (UUID).
+    """
+    storage_path = Path(config.ols_config.user_data_collection.feedback_storage)
+    feedback_file = storage_path / f"{feedback_id}.json"
+    if feedback_file.exists():
+        feedback_file.unlink()
+        if not feedback_file.exists():
+            logger.info(f"feedback file '{feedback_file}' removed")
+        else:
+            logger.error(f"feedback file '{feedback_file}' failed to remove")
+    else:
+        logger.error(f"feedback file '{feedback_file}' not found")
+
+
+@router.get("/status")
+def feedback_status() -> StatusResponse:
+    """Handle feedback status requests.
+
+    Returns:
+        Response indicating the status of the feedback.
+    """
+    logger.info("feedback status request received")
+    feedback_status = get_feedback_status()
+    return StatusResponse(functionality="feedback", status={"enabled": feedback_status})
+
+
+# TODO: OLS-136 implements the collection mechanism - revisit the need
+# of this endpoint
+# If endpoint stays in place, it needs to be properly secured - OLS-404
+@router.get("/list")
+def get_user_feedbacks(
+    feedback_is_enabled: Any = Depends(feedback_is_enabled),
+    auth: Any = Depends(auth_dependency),
+) -> FeedbacksListResponse:
+    """Handle feedback listing requests.
+
+    Args:
+        feedback_is_enabled: The feedback handler (FastAPI Depends) that
+            will handle feedback status checks.
+        auth: The Authentication handler (FastAPI Depends) that will
+            handle authentication Logic.
+
+    Returns:
+        Response containing the list of feedbacks.
+    """
+    logger.info("feedback list request received")
+
+    feedbacks = list_feedbacks()
+
+    return FeedbacksListResponse(feedbacks=feedbacks)
+
+
 @router.post("")
-def user_feedback(
-    feedback_request: FeedbackRequest, auth: Any = Depends(auth_dependency)
+def store_user_feedback(
+    feedback_request: FeedbackRequest,
+    feedback_is_enabled: Any = Depends(feedback_is_enabled),
+    auth: Any = Depends(auth_dependency),
 ) -> FeedbackResponse:
     """Handle feedback requests.
 
     Args:
         feedback_request: The request containing feedback information.
-        auth: The Authentication handler (FastAPI Depends) that will handle authentication Logic.
+        feedback_is_enabled: The feedback handler (FastAPI Depends) that
+            will handle feedback status checks.
+        auth: The Authentication handler (FastAPI Depends) that will
+            handle authentication Logic.
 
     Returns:
-        Response indicating the status of the feedback.
+        Response indicating the status of the feedback storage request.
     """
-    logger.info(
-        f"feed back received for conversation '{feedback_request.conversation_id}', "
-        f"feedback blob: {feedback_request.feedback_object}"
-    )
+    logger.info(f"feedback received {feedback_request}")
+
+    user_id = auth[0]
```

Comment:
it's used on several places. Probably time to refactor it as `auth[0]` is a bit wacky

Reply:
Do you have a story in mind?